### ISSUE TITLE
(Android) force_location_update() seems to be spamming user with timeout errors

### ISSUE THREAD

**Issue by TeMPOraL** [PebbleSeizureDetect/PebbleSeizureDetect#4]
How important is that `force_location_update()` hack? Isn't e.g. `navigator.geolocation.getCurrentPosition()` with `locationOptions.maximumAge` set to `0` sufficient?

On Android, this hacks end up popping up a GPS ERROR (code = 3, i.e. timeout) notification every 60 seconds.

Either way, it may be worth considering logging location timeout error silently. Did it pop up in practice in your experience?

**Comment by PebbleSeizureDetect** [comment]
Unfortunately it seems 100% necessary, as far as I can tell.  Other people report similar problems:

http://stackoverflow.com/questions/13386996/navigator-geolocation-getcurrentposition-not-updating-in-chrome-mobile

Though it's possible it's only needed on iOS?  I have not tested on Android.

**Comment by PebbleSeizureDetect** [comment]
Actually, the user in that Stack Overflow link was on Android.  So yeah, it may be 100% necessary.

**Comment by Waste-Land** [comment]
Thanks so much for putting this app together. I am trying to get this all setup for my wife and have been trying to figure out how to suppress the error listed in this thread. "GPS ERROR: (3) Timeout expired"

I can make it only pop up once an hour or more...but i am thinking that will mean the GPS cords it transmits will be an hour old? 

Very new to coding so trying to struggle through this. Any help would be greatly appreciated.

In a nutshell, i just want to suppress the error msg so it doesnt bug her, but leave the app fully functional.

Running this on Android

Thanks!

**Comment by PebbleSeizureDetect** [comment]
Did you try commenting out the two "force_location_update()" lines?

**Comment by Waste-Land** [comment]
I will give that a shot, won't that cause the gps location when she has a attack to be incorrect though? Like i said, my understanding of code is very limited right now lol. Thank you for the response!

**Comment by bygRyan** [comment]
I have not tested it on Android so it may work correctly even with those
lines commented out.  They were a workaround for a problem on iOS that may
or may not also affect Android.  Just make sure you test to see if the
location data is accurate after you make this change.

**Comment by Waste-Land** [comment]
Will do sir, and I will update this post on results incase anyone else has question in future. Thanks!

**Comment by Waste-Land** [comment]
OK so by commenting these two lines:

//console.warn('Location error (' + err.code + '): ' + err.message);
//Pebble.showSimpleNotificationOnPebble("SEIZURE DETECT", "GPS ERROR: (" + err.code + "): " + err.message);

I was able to suppress the GPS alerts. However, i must have screwed something up somewhere else because now when it sends an alert to my number it is only sending 0.0 for the GPS location. I have tried un-commenting those two lines and still not getting a valid GPS location. 

I pasted the current code below, if you have a chance and could see where i might have screwed up the location function i would appreciate it! 

`var locationOptions = { "timeout": 3000, "maximumAge": 60000, "enableHighAccuracy": true }; 
var lastLatitude = 0;
var lastLongitude = 0;

function locationSuccess(pos)
{
    var coordinates = pos.coords;
    console.log("Coordinates!:");
    console.log(coordinates.latitude);
    console.log(coordinates.longitude);

```
lastLatitude = coordinates.latitude;
lastLongitude = coordinates.longitude;
```

}

function locationError(err)
{
    //console.warn('Location error (' + err.code + '): ' + err.message);
    //Pebble.showSimpleNotificationOnPebble("SEIZURE DETECT", "GPS ERROR: (" + err.code + "): " + err.message);
}

function send_text_message(messageBody, phoneNumber)
{
    messageBody = "Body=" + messageBody + "+at+location:+http://maps.google.com/?q=" + lastLatitude + "," + lastLongitude + "&From=%2212089173888%22&To=%22%2B1" + phoneNumber + "%22";

```
var req = new XMLHttpRequest();
```

  req.open("POST", "https://AC342627eac7a6d584b82d93f983777a232dc6@api.twilio.com/2010-04-01/Accounts/AC342627ecc113f7a5b/SMS/Messages.json", true);
    //req.setRequestHeader("Authorization", "Basic QUMzNDI2MjdlYNmQ1ODRiODJkOTNmOTgzNzc3YTIzMmRjNg==");
    req.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
    req.setRequestHeader("Content-length", messageBody.length);
    req.setRequestHeader("Connection", "close");
    req.onload = function(e)
    {
        if ((req.readyState == 4) && ((req.status == 200) || (req.status == 201)))
        {
            console.log(req.responseText);
        }
        else
        {
            console.log("HTTP post error!  Body: " + messageBody + " Number: " + phoneNumber + " req: " + JSON.stringify(req));
            Pebble.showSimpleNotificationOnPebble("SEIZURE DETECT", "HTTP POST ERROR. See console log.");
        }
    };
    req.send(messageBody);
}

function send_location()
{
    /_var req = new XMLHttpRequest();
    req.open("GET", "http://yourwebserverdomainname.com/yourfolder/postdir.php?q=" + lastLatitude + "," + lastLongitude, true);
    req.onload = function(e)
    {
        if ((req.readyState == 4) && ((req.status == 200) || (req.status == 201)))
        {
            console.log(req.responseText);
        }
        else
        {
            console.log("HTTP post error!  Could not send location.  req: " + JSON.stringify(req));
            Pebble.showSimpleNotificationOnPebble("SEIZURE DETECT", "Failed to send location ping. See console log.");
        }
    };
    req.send();_/
}

// This is a hack needed to wake up GPS on the phone, otherwise you will get old position data
// But we don't want to leave watchPosition on all the time or it eats up battery
function force_location_update()
{
    var options = { enableHighAccuracy: true, maximumAge: 100, timeout: 60000 };
    var watchID = navigator.geolocation.watchPosition( locationSuccess, locationError, options );
    setTimeout( function() { navigator.geolocation.clearWatch( watchID ); }, 1000 );
}

Pebble.addEventListener("ready", function(e)
{
    force_location_update();
    navigator.geolocation.getCurrentPosition(locationSuccess, locationError, locationOptions);
    send_location();
});

Pebble.addEventListener("appmessage", function(e)
{
    console.log('Phone received message: ' + e.payload["0"]);
    force_location_update();
    navigator.geolocation.getCurrentPosition(locationSuccess, locationError, locationOptions);

```
 //Panic message?
if (e.payload["0"] == 1)
{
    send_text_message("MANDY+pushed+the+panic+button", "509761");
    //send_text_message("MANDY+pushed+the+panic+button", "7785556789");
}
 //Seizure message?
else if (e.payload["0"] == 2)
{
    send_text_message("MANDY+may+have+had+a+seizure", "509761");
    //send_text_message("MANDY+may+have+had+a+seizure", "7785556789");
}
 //Location ping?
else if (e.payload["0"] == 3)
{
    send_location();
}
```

});`

**Comment by PebbleSeizureDetect** [comment]
Can you try going back to the original source code, and then commenting out
only the lines that contain "force_location_update()"?
